services/drupal/scripts/ecs/drush-migrate.sh

```diff
@@ -42,13 +43,30 @@ run_migration() {
       --limit=$batch_size \
       --continue-on-failure \
       "$migration"
+
+    # Check if we've asked to halt. Migration's own status field isn't really reliable
+    # for discerning whether a migration has been stopped or completed.
+    halted=$(drush state-get epa.migrations_halted)
+    if test -n "$halted"; then
+      # We check again and exit below
+      break
+    fi
   done
 
+  # Calculate the difference in seconds for some timing statistics
   finish="$(date +%s)"
+  time=$((finish - start))
 
   # Determine how many unprocessed items were left behind by the migration.
   unprocessed="$(drush ms "$migration" --field=unprocessed)"
 
+  if test -n "$halted"; then
```

Comment:
What's the reasoning for the second check here? Why not return directly in the loop?

Reply:
I wanted to output the unprocessed.